### DISCUSSION TITLE
"Unary" and "pass through" nodes.

### DISCUSSION THREAD

**Discussion by hyanwong** [tskit-dev/what-is-an-arg-paper#278] [General]
At the moment we use the term "unary node" to mean a node that is unary in all the local trees, but which need not be unary in the graph itself (e.g. it has different parents in different regions). We use the term "pass though" to mean a node that is also unary in the graph itself (one child and one parent, everywhere in the genome).

I wonder if the term "unary node" is confusing to people coming from the graph world, as e.g. a CA-non-coalescent node certainly doesn't look unary when plotting the graph. Do we need an extra adjective in there, like "unary tree node", or "unary branch node", or even something different like "non-branching node"?

(I use the term "branch" here because we have tended to use "branch" to refer to a connection in a single tree, whereas "edge" means a connection that can span multiple trees.)

**Comment by JereKoskela** [comment]
To me, "unary tree node" sounds disconnected from the graph outright, maybe as something which is only present / relevant when restricting to a tree. I think I'd be in favour of just explaining what we mean by "unary" and "pass through" carefully, but I don't feel strongly either way.
